hw3/01-routing.js

```diff
@@ -34,13 +22,62 @@ const server = http.createServer((req, res) => {
     return result;
   };
 
-  if (req.url === '/') {
-    let routeResults = getRoutes();
-
-    res.writeHead(200, { 'Content-Type': 'text/html' });
-    res.write(`<h1>Exercise 01</h1>`);
-    res.write(`<ul> ${routeResults} </ul>`);
-    res.end();
+  //Handling routing with switch statement- I think this is more readable than an extended if/else structure.
+  switch (req.url) {
+    case "/":
+      let routeResults = getRoutes();
+      res.writeHead(200, { "Content-Type": "text/html" });
+      res.write(`<h1>Exercise 01</h1>`);
+      res.write(`<ul> ${routeResults} </ul>`);
+      res.end();
+      break;
+    case "/welcome":
+      res.writeHead(200, { "Content-Type": "text/html" });
+      res.write(`<h1>Welcome!</h1>`);
+      res.end();
+      break;
+    case "/redirect":
+      res.writeHead(302, { Location: "/redirected" });
+      res.end();
+      break;
+    case "/redirected":
+      res.writeHead(200, { "Content-Type": "text/html" });
+      res.write(`<h1>You have been redirected!</h1>`);
+      res.write(
+        `Unless you clicked the redirected link on the main page. Then you just travelled here.`
+      );
+      res.end();
+      break;
+    case "/cache":
+      res.writeHead(
+        200,
+        { "Content-Type": "text/html" },
+        { "Cache-Control": "max-age: 86_400" }
```

Comment:
The underscore in 86_400 is wrong.

Reply:
Not from a syntax standpoint.  Might be from a style standpoint, but it makes numbers so much more readable that I'm willing to take the hit.